packages/web/server/api/edge-routers/pools-router.ts

```diff
@@ -108,59 +109,68 @@ export const poolsRouter = createTRPCRouter({
-            ]);
+                /** Get remote data via concurrent requests, if needed. */
+                const [pools, incentives, marketMetrics] = await Promise.all([
+                  poolsPromise,
+                  incentivesPromise,
+                  marketMetricsPromise,
+                ]);
 
-            const marketIncentivePools = pools
-              .map((pool) => {
-                const incentivesForPool = incentives.get(pool.id);
-                const metricsForPool = marketMetrics.get(pool.id) ?? {};
+                const marketIncentivePools = pools
+                  .map((pool) => {
+                    const incentivesForPool = incentives.get(pool.id);
+                    const metricsForPool = marketMetrics.get(pool.id) ?? {};
 
-                const isIncentiveFiltered =
-                  incentivesForPool &&
-                  isIncentivePoolFiltered(incentivesForPool, {
-                    incentiveTypes,
-                  });
+                    const isIncentiveFiltered =
+                      incentivesForPool &&
+                      isIncentivePoolFiltered(incentivesForPool, {
+                        incentiveTypes,
+                      });
 
-                if (isIncentiveFiltered) return;
+                    if (isIncentiveFiltered) return;
 
-                return {
-                  ...pool,
-                  ...incentivesForPool,
-                  ...metricsForPool,
-                };
-              })
-              .filter((pool): pool is NonNullable<typeof pool> => !!pool);
+                    return {
+                      ...pool,
+                      ...incentivesForPool,
+                      ...metricsForPool,
+                    };
+                  })
+                  .filter((pool): pool is NonNullable<typeof pool> => !!pool);
 
-            // won't sort if searching
-            if (search) return marketIncentivePools;
-            else
-              return sort(
-                marketIncentivePools,
-                sortInput.keyPath,
-                sortInput.direction
-              );
-          },
-          cacheKey: JSON.stringify({
-            search,
-            sortInput,
-            minLiquidityUsd,
-            types,
-            incentiveTypes,
-          }),
-          cursor,
-          limit,
-        })
+                // won't sort if searching
+                if (search) return marketIncentivePools;
+                else
+                  return sort(
+                    marketIncentivePools,
+                    sortInput.keyPath,
+                    sortInput.direction
+                  );
+              },
+              cacheKey: JSON.stringify({
+                search,
+                sortInput,
+                minLiquidityUsd,
+                types,
+                incentiveTypes,
+              }),
+              cursor,
+              limit,
+            }),
+          12_000,
+          "getMarketIncentivePools"
```

Comment:
Why wrap pagination in timeout? Does this alleviate edge memory issues?

Reply:
Although it doesn't solve the issue directly, it allows the function to fail quickly, freeing up resources for other requests. Additionally, this enables the Front-End to retry the query with a different function in case of a timeout

Reply:
Is this common enough to consider adding the timeout inside of `maybeCachePaginatedItems`? Perhaps we can add an optional `timeoutName` param to make the timeout optional?

Reply:
I believe it's a good idea to abstract this behavior. However, instead of increasing the complexity of `maybeCachePaginatedItems`, how about we create a `timeoutProcedure` within trpc so all routes can share a timeout? This would also give us visibility in the logs for all routes since timed-out functions return an error and status code.

Reply:
Is that like a global tRPC timeout?